packages/foundry/src/vault/adapter/aave/aaveV2/AaveV2Adapter.sol

```diff
@@ -0,0 +1,146 @@
+
+import { AdapterBase, IERC20, IERC20Metadata, SafeERC20, ERC20, Math, IStrategy, IAdapter } from "../../abstracts/AdapterBase.sol";
+import { WithRewards, IWithRewards } from "../../abstracts/WithRewards.sol";
+import { ILendingPool, IAaveMining, IAToken } from "./IAaveV2.sol";
+
+/**
+ * @title   AaveV2 Adapter
+ * @author  amatureApe
+ * @notice  ERC4626 wrapper for AaveV2 Vaults.
+ *
+ * An ERC4626 compliant Wrapper for https://github.com/aave/protocol-v2/blob/master/contracts/protocol/lendingpool/LendingPool.sol.
+ * Allows wrapping AaveV2 aTokens with or without an active Liquidity Mining.
+ * Allows for additional strategies to use rewardsToken in case of an active Liquidity Mining.
+ */
+
+contract AaveV2Adapter is AdapterBase, WithRewards {
+  using SafeERC20 for IERC20;
+  using Math for uint256;
+
+  string internal _name;
+  string internal _symbol;
+
+  // @notice The Aave aToken contract
+  IAToken public aToken;
+
+  // @notice The Aave liquidity mining contract
+  IAaveMining public aaveMining;
+
+  // @notice Check to see if Aave liquidity mining is active
+  bool public isActiveMining;
+
+  // @notice The Aave LendingPool contract
+  ILendingPool public lendingPool;
+
+  /*//////////////////////////////////////////////////////////////
+                            CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+  /**
+   * @notice Initialize a new AaveV2 Adapter.
+   * @param adapterInitData Encoded data for the base adapter initialization.
+   * @param _aToken Aave wrapped asset. Can also be used to get lendingPool and aaveMining.
+   * @dev `_aToken` - The underlying asset supplied to and wrapped by Aave.
+   * @dev `_lendingPool` - The lending pool.
+   * @dev `_aaveMining` - An optional liquidity mining contract to boost yield.
+   * @dev This function is called by the factory contract when deploying a new vault.
+   */
+  function initialize(
+    bytes memory adapterInitData,
+    address,
+    address _aToken
+  ) public {
+    __AdapterBase_init(adapterInitData);
+
+    _name = string.concat("Popcorn AaveV2", IERC20Metadata(asset()).name(), " Adapter");
+    _symbol = string.concat("popB-", IERC20Metadata(asset()).symbol());
+
+    aToken = IAToken(_aToken);
+    lendingPool = ILendingPool(aToken.POOL());
+    aaveMining = IAaveMining(aToken.getIncentivesController());
+
+    IERC20(asset()).approve(address(lendingPool), type(uint256).max);
+
+    uint256 emission;
+    if (address(aaveMining) != address(0)) {
+      (, emission, ) = aaveMining.getAssetData(asset());
+    }
+
+    isActiveMining = emission > 0 ? true : false;
+  }
+
+  /*//////////////////////////////////////////////////////////////
+                            ACCOUNTING LOGIC
+  //////////////////////////////////////////////////////////////*/
+
+  function totalAssets() public view override returns (uint256) {
+    return paused() ? IERC20(asset()).balanceOf(address(this)) : aToken.balanceOf(address(this));
```

Comment:
I think we need to use `scaledBalanceOf` instead of just `balanceOf` on the aToken

Reply:
This makes sense to me. I've already changed it, but just a heads up that the Timeless Fi example used balanceOf https://github.com/timeless-fi/yield-daddy/blob/main/src/aave-v2/AaveV2ERC4626.sol#L136-L139